Modules/xSharePoint/DSCResources/MSFT_xSPAntivirusSettings/MSFT_xSPAntivirusSettings.psm1

```diff
@@ -0,0 +1,113 @@
+        [parameter(Mandatory = $false)] [System.Boolean] $AllowDownloadInfected,
+        [parameter(Mandatory = $false)] [System.Boolean] $AttemptToClean,
+        [parameter(Mandatory = $false)] [System.UInt16] $TimeoutDuration,
+        [parameter(Mandatory = $false)] [System.UInt16] $NumberOfThreads,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    Write-Verbose -Message "Getting antivirus configuration settings"
+
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        
+        [System.Reflection.Assembly]::LoadWithPartialName("Microsoft.SharePoint") | Out-Null
+
+        # Get a reference to the Administration WebService
+        $admService = [Microsoft.SharePoint.Administration.SPWebService]::ContentService
+
+        if ($null -eq $admService) { return $null }
+        
+        return @{
+            # Set the antivirus settings
+            AllowDownloadInfected = $admService.AntivirusSettings.AllowDownload
+            ScanOnDownload = $admService.AntivirusSettings.DownloadScanEnabled
+            ScanOnUpload = $admService.AntivirusSettings.UploadScanEnabled
+            AttemptToClean = $admService.AntivirusSettings.CleaningEnabled
+            NumberOfThreads = $admService.AntivirusSettings.NumberOfThreads
+            TimeoutDuration = $admService.AntivirusSettings.Timeout.TotalSeconds
+            InstallAccount = $params.InstallAccount
+        }
+    }
+
+    return $result
+}
+
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory = $true)] [System.Boolean] $ScanOnDownload,
+        [parameter(Mandatory = $false)] [System.Boolean] $ScanOnUpload,
+        [parameter(Mandatory = $false)] [System.Boolean] $AllowDownloadInfected,
+        [parameter(Mandatory = $false)] [System.Boolean] $AttemptToClean,
+        [parameter(Mandatory = $false)] [System.UInt16] $TimeoutDuration,
+        [parameter(Mandatory = $false)] [System.UInt16] $NumberOfThreads,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    Write-Verbose -Message "Setting antivirus configuration settings"
+
+    Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        
+        Write-Verbose -Message "Start update"
+        # Load the SharePoint Assembly, using old style for backward compatibility with V1
+        [System.Reflection.Assembly]::LoadWithPartialName("Microsoft.SharePoint") | Out-Null
+
+        # Get a reference to the Administration WebService
+        $admService = [Microsoft.SharePoint.Administration.SPWebService]::ContentService
+
+        # Set the antivirus settings
+        if ($params.ContainsKey("AllowDownloadInfected")) { 
+            Write-Verbose -Message "Setting Allow Download"
```

Comment:
When we run things inside of our Invoke-xSharePointCommand though we get the output

    Invoke-xSharePointCommand -ScriptBlock { Write-Verbose "this is a test" } 

Now this may only be happening when we call our cmdlet without a credential (which is how it runs with PowerShell 5's PsDscRunAsCredential parameter instead of the InstallAccount we built in which will use a session like you tested with). Given the plan is to prefer the use of the PowerShell 5 way of running things, we can drop verbose output in to our script and see it that way.